integrations/access/servicenow/client.go

```diff
@@ -224,15 +224,15 @@ func (snc *Client) GetOnCall(ctx context.Context, rotaID string) ([]string, erro
 	if len(result.Result) == 0 {
 		return nil, trace.NotFound("no user found for given rota: %q", rotaID)
 	}
-	var emails []string
+	var userNames []string
 	for _, result := range result.Result {
-		email, err := snc.GetUserEmail(ctx, result.UserID)
+		userName, err := snc.GetUserName(ctx, result.UserID)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
-		emails = append(emails, email)
+		userNames = append(userNames, userName)
 	}
-	return emails, nil
+	return userNames, nil
```

Comment:
`GetOnCall()` now returns the username instead of the user email. This means the `userIsOnCall` now compares the Teleport user with the servicenow username instead of the servicenow email.

This seems like a breaking change. Was it a bug? What is the expected behaviour, checking the username or the email? If we want to do this breaking change we might want to add something in the release notes.

Reply:
This function was not actually called prior to this PR and some changes in [2503](https://github.com/gravitational/teleport.e/pull/2503)